aggregate/gcp_billing/main.py

```diff
@@ -12,16 +12,75 @@
 - Service ID should be faithfully handed over
 """
 
+import json
 
-def main(request):
+from google.cloud import bigquery
 
-    start_period = None
-    finish_period = None
-    pass
+from cpg_utils.cloud import read_secret
+
+
+ANALYSIS_RUNNER_PROJECT_ID = 'analysis-runner'
+SOURCE_TABLE = 'billing-admin-290403.billing.gcp_billing_export_v1_01D012_20A6A2_CBD343'
+DESTINATION_TABLE = 'sabrina-dev-337923.billing.aggregate'
+
+bigquery_client = bigquery.Client()
+
+
+def main():
+    """Main entry point for the Cloud Function"""
+    start_period = "2022-04-05"
+    finish_period = "2022-04-05"
+
+    # Get the dataset to GCP project map
+    dataset_to_gcp_map = get_dataset_to_gcp_map()
+    allowed_project_ids = "'" + "','".join(dataset_to_gcp_map.values()) + "'"
+
+    # Get the billing date in the time period
+    # Filter out any rows that aren't in the allowed project ids
+    _query = f"""
+        SELECT * FROM `{SOURCE_TABLE}`
+        WHERE DATE(usage_end_time) >= '{start_period}'
+            AND DATE(usage_end_time) <= '{finish_period}'
```

Comment:
Only one of these should be equal, as you risk grabbing a record sat right on the changeover

Reply:
Sorted. I made it start date inclusive